### ISSUE TITLE
Background instant fading on fail looks ridiculous with light backgrounds

### ISSUE THREAD

**Issue by shsh-x** [ppy/osu#30287]
### Type

Cosmetic

#### Description
When failing on a map with a light BG, the bg fades out to black in one frame

#### Expected behavior
Gradual fading out of bg on fail (and not to pitch black)

#### How to reproduce
1. Create a map with white bg
2. Fail it
3. Look at the painful fade

(Actually it may be related to the light UI theme feature requests since the game is not designed to play with light bgs or negative dimming (yet?))

### Screenshots or videos

https://github.com/user-attachments/assets/fdcd24f2-e72a-47c0-940b-49e24901a338

### Version

2024.1009.1-lazer

### Logs

[compressed-logs.zip](https://github.com/user-attachments/files/17368286/compressed-logs.zip)

**Comment by smoogipoo** [comment]
Please provide the/a beatmap.

**Comment by peppy** [comment]
Also the skin.

**Comment by shsh-x** [comment]
Honestly, I don't remember which beatmap I was recording it on, but it happens on any beatmap with white bg.
And about the skin, since I don't have a supporter tag, the background is unrelated here, and also the skin doesn't have any legacy fail screens as I'm developing it for lazer only.

But I'll try to create a beatmap by myself to check and attach it here if the issue will persist on a fresh beatmap in less than a day.

**Comment by peppy** [comment]
Seems like something may have regressed in that case. I can confirm it's looking weird.

**Comment by shsh-x** [comment]
[failscreen testing - null (shshx).zip](https://github.com/user-attachments/files/17384606/failscreen.testing.-.null.shshx.zip)
Here's the map to test bg. Bg dim to 0% and just start it.
Also don't forget to change .zip to .olz :p

**Comment by shsh-x** [comment]
Also tested it on all default skins and now I'm sure it's definitely not related to skinning at all.
After enabling "Fade playfield to red when health is low" the transition looks a little bit smoother but it obviously doesn't fix the issue since insta fade is still visible.

**Comment by peppy** [comment]
I started [looking into this](https://github.com/ppy/osu/compare/master...peppy:osu:background-on-fail-screen?expand=1) just to get a feel for how it would look visually if fixed. Not sure yet.

**Comment by shsh-x** [comment]
If you'd ask my opinion on how it'd look better, I'd say it'll be better to check average color of bg and make smooth transition to black or white through this color, but ofc if you got no plans about light theming the game, the only option left is transition to black through the average of bg
OR
The easier to implement is just make a fade animation of fail screen bg with the end point of opacity at 60-80%.
Like when you fail the steps for fail screen to appear are: 0% opacity full screen black box appears > start opacity change animation up to 80% > (somewhere at the middle of previous animation(depends on the length)) start appearing fail screen buttons > end of animations > voila

**Comment by peppy** [comment]
Yes I understand how to transition. It's more so the dual background layers and the code structure which I am pondering on.
